Rakefile

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+docker_run_prefix = "docker run --tty --rm --volume #{Dir.pwd}:/usr/src/app --workdir /usr/src/app cfn-nag-dev:latest"
+
+def ensure_local_dev_image
+  puts 'Checking for cfn-nag-dev Docker image...'
+  image = `docker images -q cfn-nag-dev:latest`
+
+  if image.empty?
+    puts 'cfn-nag-dev image is missing. Building it first...'
+    Rake::Task['build_docker_dev'].invoke
+  else
+    puts 'cfn-nag-dev image exists, proceeding.'
+  end
+end
+
+desc 'Build the local Docker iamge for development/testing'
+task :build_docker_dev do
+  sh 'docker build --file Dockerfile-dev --rm --tag cfn-nag-dev .'
+end
+
+namespace 'test' do
+  desc 'Run all rspec tests'
+  task :all do
+    ensure_local_dev_image
+    sh "#{docker_run_prefix} ./scripts/rspec.sh"
```

Comment:
it makes sense to me to run e2e tests in a docker container in an installed gem... it seems counter-intuitive to me to run my unit tests against an installed gem vs just running them in my dev environment against the actual source base... i mean i guess it's not wrong?  just seems like extra over what i need to get fast feedback during development

Reply:
you could still get fast feedback. everything just runs in a local container vs directly on your machine. The idea was to not have to install anything locally (in case i'm working on several projects where i may have several versions of ruby, python, node, etc required). The commands/tests should be exactly the same though